app/lib/katello/repo_discovery.rb

```diff
@@ -4,203 +4,17 @@ module Katello
   class RepoDiscovery
     include Katello::Util::HttpProxy
 
-    attr_reader :found, :crawled, :to_follow
-
-    # rubocop:disable Metrics/ParameterLists
-    def initialize(url, content_type = 'yum', upstream_username = nil,
-      upstream_password = nil, search = '*', crawled = [],
-      found = [], to_follow = [])
-      @uri = uri(url)
-      @content_type = content_type
-      @upstream_username = upstream_username.empty? ? nil : upstream_username
-      @upstream_password = upstream_password.empty? ? nil : upstream_password
-      @search = search
-      @found = found
-      @crawled = crawled
-      @to_follow = to_follow
+    def self.class_for(content_type = 'yum')
+      repo_discovery_class = RepositoryTypeManager.find_repository_type(content_type)&.repo_discovery_class
+      fail _ "Content type does not support repo discovery" unless repo_discovery_class
```

Comment:
is `_` a method call? If so, I'd prefer braces because it's very easy to miss that it's a method call
```suggestion
      fail _("Content type does not support repo discovery") unless repo_discovery_class
```